File: sdk/csharp/Test/Nem/Crypto/AddressTest.cs
using Newtonsoft.Json;
using SymbolSdk;
using SymbolSdk.Nem;

namespace Test.Nem.Crypto;
public class AddressTest
{
	private readonly NemFacade MainFacade = new (Network.MainNet);
    private readonly NemFacade TestFacade = new (Network.TestNet);
    
    [Test]
    public async Task Address()
    {
	    var file = new FileInfo("../../../../../../tests/vectors/nem/crypto/1.test-address.json");
	    var contents = await File.ReadAllTextAsync(file.FullName);
	    var jsonMap = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(contents);

	    if (jsonMap != null)
		    foreach (var t in jsonMap)
            {
                var publicKey = new PublicKey((string)t["publicKey"]);
			    var mainNetwork = Network.MainNet;
			    var testNetwork = Network.TestNet;
			    var mainAddress = mainNetwork.PublicKeyToAddress(publicKey).ToString();
			    var testAddress = testNetwork.PublicKeyToAddress(publicKey).ToString();
                Assert.Multiple(() =>
                {
                    Assert.That(mainAddress, Is.EqualTo((string)t["address_Public"]));
                    Assert.That(testAddress, Is.EqualTo((string)t["address_PublicTest"]));
                });
            }
    }
}

File: sdk/php/tests/symbol/crypto/AddressTest.php
<?php

namespace SymbolSdk\Symbol\Models;

require_once __DIR__ . '/../../../src/symbol/models.php';

use PHPUnit\Framework\TestCase;
use SymbolSdk\Symbol\Network;

use function SymbolSdk\Utils\binaryToAddress;

class AddressTest extends TestCase
{
  public function testAddress()
  {
    $jsonFilePath = '/Users/matsukawatoshiya/programs/symbol-sdk/symbol/tests/vectors/symbol/crypto/1.test-address.json';
    $jsonData = file_get_contents($jsonFilePath);
    $decodedData = json_decode($jsonData, true);

    foreach ($decodedData as $item) {
      $publicKey = new PublicKey($item['publicKey']);
      Network::initialize();
      $mainNetwork = Network::$MAINNET;
      $testNetwork = Network::$TESTNET;
      $mainAddress = $mainNetwork->publicKeyToAddress($publicKey->binaryData);
      $testAddress = $testNetwork->publicKeyToAddress($publicKey->binaryData);

      $this->assertEquals($mainAddress, $item['address_Public']);
      $this->assertEquals($testAddress, $item['address_PublicTest']);
    }
  }
}
